io/plugins/eu.esdihumboldt.hale.io.deegree.ui/src/eu/esdihumboldt/hale/io/deegree/ui/BasicMappingConfigurationPage.java

```diff
@@ -0,0 +1,383 @@
+		MappingMode mode = GenericMappingConfiguration.DEFAULT_MAPPING_MODE;
+		if (!modeSel.isEmpty() && modeSel instanceof IStructuredSelection) {
+			mode = (MappingMode) ((IStructuredSelection) modeSel).getFirstElement();
+		}
+		mappingConfig.setMappingMode(mode);
+
+		mappingConfig.setMaxNameLength(Optional.of(nameLength.getSelection()).filter(x -> x > 0));
+
+		mappingConfig.setUseNamespacePrefixForTableNames(useNamespacePrefix.getSelection());
+
+		mappingConfig.setUseIntegerIDs(useIntegerIDs.getSelection());
+
+		// CRS
+
+		mappingConfig.setCRSIdentifier(crsRef.getText());
+
+		mappingConfig.setDimension(Optional.of(dimension.getSelection()).filter(x -> x > 0));
+
+		mappingConfig.setSRID(Optional.ofNullable(srid.getText()).filter(x -> !x.isEmpty()));
+
+		// overall
+
+		ProviderConfig.set(mappingConfig.getInternalConfig(), provider);
+
+		return true;
+	}
+
+	@Override
+	protected void createContent(Composite page) {
+		page.setLayout(new GridLayout(1, false));
+		GridDataFactory groupData = GridDataFactory.fillDefaults().grab(true, false);
+
+		GridDataFactory defLabel = GridDataFactory.swtDefaults().align(SWT.END, SWT.CENTER);
+		GridDataFactory longField = GridDataFactory.swtDefaults().align(SWT.FILL, SWT.CENTER)
+				.span(2, 1).grab(true, false);
+
+		// Database group
+
+		Group database = new Group(page, SWT.NONE);
+		database.setLayout(new GridLayout(3, false));
+		database.setText("Database");
+		groupData.applyTo(database);
+
+		// connection ID
+		Label connIdLabel = new Label(database, SWT.NONE);
+		connIdLabel.setText("Connection ID");
+		defLabel.applyTo(connIdLabel);
+		connId = new Text(database, SWT.SINGLE | SWT.BORDER);
+		connId.addModifyListener(new ModifyListener() {
+
+			@Override
+			public void modifyText(ModifyEvent e) {
+				updateState();
+			}
+		});
+		longField.applyTo(connId);
+
+		// database type
+		Label dbTypeLabel = new Label(database, SWT.NONE);
+		dbTypeLabel.setText("Database type");
+		defLabel.applyTo(dbTypeLabel);
+		dbType = new ComboViewer(database);
+		dbType.setContentProvider(EnumContentProvider.getInstance());
+		dbType.setInput(DatabaseType.class);
+		longField.applyTo(dbType.getControl());
+
+		// and version
+		Label dbVersionLabel = new Label(database, SWT.NONE);
+		dbVersionLabel.setText("Database version");
+		defLabel.applyTo(dbVersionLabel);
+		dbVersion = new Text(database, SWT.SINGLE | SWT.BORDER);
+		longField.applyTo(dbVersion);
+
+		// mapping mode
+		Label modeLabel = new Label(database, SWT.NONE);
+		modeLabel.setText("Mapping mode");
+		defLabel.applyTo(modeLabel);
+		mode = new ComboViewer(database);
+		mode.setContentProvider(EnumContentProvider.getInstance());
+		mode.setInput(MappingMode.class);
+		longField.applyTo(mode.getControl());
+
+		// max name length
+		Label nameLengthLabel = new Label(database, SWT.NONE);
+		nameLengthLabel.setText("Max name length");
+		defLabel.applyTo(nameLengthLabel);
+		nameLength = new Spinner(database, SWT.NONE);
```

Comment:
It looks a little nicer on Windows when passing `SWT.BORDER` here:
![grafik](https://user-images.githubusercontent.com/1867607/43471531-8352e886-94eb-11e8-80fc-62909729f2a6.png)

Didn't check what effect that has on Linux, though.

Reply:
I was also wondering. The `Spinner` doc only mentioned *READ_ONLY* and *WRAP* for styles, so I skipped it. But if it is supported it makes sense to add it.